src/proto/nc_redis.c

```diff
@@ -1633,7 +1633,13 @@ redis_parse_req(struct msg *r)
 
             m = p + r->rlen;
             if (m >= b->last) {
-                r->rlen -= (uint32_t)(b->last - p);
+                /* For EVAL/EVALHASH, the r->token has been assigned a value. When 
+                 * m > b->last will due to repair mbuf, so only r->token == NULL, 
+                 * need caculate r->rlen again. */
+                if (r->token == NULL)
+                {
```

Comment:
```suggestion
                /* 
                 * For EVAL/EVALHASH, the r->token has been assigned a value. When 
                 * m >= b->last happens we will need to repair mbuf. So, when r->token == NULL, 
                 * we will need to calculate r->rlen again.
                 *
                 * At the end of redis_parse_req, r->token will be used to choose the start (p) for the next call to redis_parse_req and clear r->token when repairing this and adding more data.
                 */
                if (r->token == NULL) {
```

Looks good to me, suggestion should make this follow https://github.com/twitter/twemproxy/blob/master/notes/c-styleguide.txt

Reply:
thanks, yours looks more better and clear!